### PR TITLE
CVE-2016-3092

### PR DESCRIPTION
**Pull request by aldenreed**

### CODE REVIEW

File: cves/CVE-2016-3092.yml [hunk 1]
```diff
@@ -43,37 +43,32 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  The system doesn't check the length of a boundary string. This means that
```
Comment (Pearse4004) [inline cves/CVE-2016-3092.yml]: What part of the system doesn't check for the boundary length?

File: cves/CVE-2016-3092.yml [hunk 2]
```diff
@@ -206,6 +211,11 @@ mistakes:
...
     Use those questions to inspire your answer. Don't feel obligated to answer
-    every one. Write a thoughtful entry here that those ing the software
+    every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    I think this was a coding mistake because, when getting string input, it is good
+    practice to check the length of the string to make sure that it won't cause any
+    problems. I think the fix looks proper because it sets the buffer size to the
+    boundary length times two if the boundary string is too large. This helps to prevent
```
Comment (Pearse4004) [inline cves/CVE-2016-3092.yml:223]: Isn't a check of the input for size and increasing buffer size overkill?

File: cves/CVE-2016-3092.yml (hunk 1 above)
Comment (aldenreed) [reply cves/CVE-2016-3092.yml]: It's the multipart stream part of the system. I will add that to the description.

File: cves/CVE-2016-3092.yml (hunk 2 above)
Comment (aldenreed) [reply cves/CVE-2016-3092.yml:223]: This is a good point. I will address this when making my edits.

Comment (aldenreed) [reply cves/CVE-2016-3092.yml:223]: Looking back on it, I think that the error can be caused by either a boundary string that is too long or one that is just smaller than the buffer size. The buffer size increase only seems to happen if the boundary string is just below the buffer size, so I don't think having the fix for both is too overkill.

File: cves/CVE-2016-3092.yml [hunk 3]
```diff
@@ -145,12 +142,17 @@ interesting_commits:
...
     section by explaining what happened between the VCCs and the fix.
-  answer:
   commits:
-    - commit:
-      note:
-    - commit:
-      note:
+    - commit: bcefe58374c421c9431c144938e2e91d22f5061b
+      note: |
+        This commit was interesting because it made some major changes to how the
```
Comment (allisonsuarez) [inline cves/CVE-2016-3092.yml:151]: Maybe a good solution would have been to not append input directly in the first place?

File: cves/CVE-2016-3092.yml [hunk 4]
```diff
@@ -43,37 +43,32 @@ description_instructions: |
...
-   - commit: 2c3553f3681baf775c50bb0b49ea61cb44ea914f
-     note: SVN rev 1743722, from the Tomcat website.
-   - commit: d752a415a875e888d8c8d0988dfbde95c2c6fb1d
-     note: SVN rev 1743738, from the Tomcat website.
-   - commit: 8999f8243197a5f8297d0cb1a0d86ed175678a77
-     note: SVN rev 1743742, from the Tomcat website.
-   - commit:
-     note:
+     note: |
+       Prevents the user from making a boundary string that is just shorter than the
```
Comment (allisonsuarez) [inline cves/CVE-2016-3092.yml:67]: Maybe add the commit where the "too long" boundary string case is fixed

**Comment by allisonsuarez** [comment]
👍

File: cves/CVE-2016-3092.yml (hunk 4 above)
Comment (aldenreed) [reply cves/CVE-2016-3092.yml:67]: Sounds good. I'll add that.

File: cves/CVE-2016-3092.yml (hunk 3 above)
Comment (aldenreed) [reply cves/CVE-2016-3092.yml:151]: This is an interesting thought. I will address this in my edit.

File: cves/CVE-2016-3092.yml [hunk 5]
```diff
@@ -10,21 +10,21 @@ CWE_instructions: |
...
   set to true.
-curated: false
+curated: true
 
 reported_instructions: |
   Was there a date that this vulnerability was reported to the team? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE data.
   Please enter your date in YYYY-MM-DD format.
-reported:
+reported: 2016-05-09
```
Comment (Nitepone) [inline cves/CVE-2016-3092.yml:27]: Genuinely curious curious where you got this date. The earliest I found was the publish date from Apache mailing list of 06/23/16 https://mail-archives.apache.org/mod_mbox/www-announce/201606.mbox/%3C45A20804-ABFF-4FED-A297-69AC95AB9A3F@apache.org%3E

Worried I may be overlooking some site from Apache that shows the actual report date to the team.

Comment (aldenreed) [reply cves/CVE-2016-3092.yml:27]: I got it from here: https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.0.M8

**Comment by Nitepone** [comment]
I give this 4 upvotes

**Pull request merged**